listeners.py

```diff
@@ -178,6 +178,55 @@ def __del__(self):
                 pass
 
 
+class ResourcesWatcher(BaseWatcher):
+    def __init__(self):
+        self.failed_tasks = dict()
+        self.rss_results = dict()
+
+    def process_result(self, obj):
+        # called only once on task run finished
+        if isinstance(obj, WorkerTaskResult):
+            task_name = obj.task_id[0] + ((':' + obj.task_id[1]) if obj.task_id[1] else '')
+            if obj.short_status == 'fail':
+                self.failed_tasks[task_name] = 1
+
+    def set_procs_rss(self, rss):
+        self.rss_results = rss
+
+    # called only once after all tasks finished
+    def print_statistics(self):
+        top_rss = 10
+        results_sorted = []
+
+        # prepare standalone subpath '<vardir>/statistics' for statistics files
+        stats_dir = os.path.join(Options().args.vardir, 'statistics')
+        safe_makedirs(stats_dir)
+
+        # RSS
+
+        # print to stdout RSS statistics for all failed tasks
+        if self.failed_tasks:
+            color_stdout('RSS of the failed tasks:\n', schema='test_var')
+            for task_name in self.failed_tasks:
+                if task_name in self.rss_results:
+                    color_stdout('* %6.1f %s\n' % (self.rss_results[task_name] / 1000, task_name),
+                                 schema='test_var')
+
+        # print to stdout RSS statistics for some number of most it used tasks
+        color_stdout('Up to {} most RSS used tasks in Mb:\n'.format(top_rss), schema='test_var')
+        results_sorted = sorted(self.rss_results.items(), key=lambda x: x[1], reverse=True)
+        for task_name, rss in results_sorted[:top_rss]:
+            color_stdout('* %6.1f %s\n' % (int(rss) / 1000, task_name), schema='test_var')
```

Comment:
1. 1000 → 1024.
2. In Python 2 it is integer division.

Reply:
Sure, with:
```
diff --git a/listeners.py b/listeners.py
index 3ab8fd4..5e5db70 100644
--- a/listeners.py
+++ b/listeners.py
@@ -237,14 +237,14 @@ class ResourcesWatcher(BaseWatcher):
             color_stdout('RSS of the failed tasks:\n', schema='test_var')
             for task_name in self.failed_tasks:
                 if task_name in self.rss_results:
-                    color_stdout('* %6.1f %s\n' % (self.rss_results[task_name] / 1000, task_name),
-                                 schema='test_var')
+                    color_stdout('* %6.1f %s\n' % (float(self.rss_results[task_name] / 1024),
+                                 task_name), schema='test_var')
 
         # print to stdout RSS statistics for some number of most it used tasks
         color_stdout('Up to {} most RSS used tasks in Mb:\n'.format(top_rss), schema='test_var')
         results_sorted = sorted(self.rss_results.items(), key=lambda x: x[1], reverse=True)
         for task_name, rss in results_sorted[:top_rss]:
-            color_stdout('* %6.1f %s\n' % (int(rss) / 1000, task_name), schema='test_var')
+            color_stdout('* %6.1f %s\n' % (float(int(rss) / 1024), task_name), schema='test_var')
         color_stdout('-' * 41, "\n", schema='separator')
 
         # print RSS statistics to '<vardir>/statistics/rss.log' file
```

Reply:
Please, don't write code in a hurry. I'm sure you able to find mistakes if you'll look again at the code:

1. `float(self.rss_results[task_name] / 1024)`
2. `float(int(rss) / 1024)`

Reply:
Right, corrected.